.github/workflows/ci.yml

```diff
@@ -0,0 +1,104 @@
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          override: true
+          toolchain: stable
+          profile: minimal
+      # Smart caching for Rust projects.
+      # Includes workaround for macos cache corruption.
+      # - https://github.com/rust-lang/cargo/issues/8603
+      # - https://github.com/actions/cache/issues/403
+      - uses: Swatinem/rust-cache@v1
+
+      - name: Build workspace
+        run: cargo build
+
+      # Workspace tests
+      - name: Run workspace unit tests
+        run: cargo test --lib --all -j6
+      - name: Run workspace doc tests
+        run: cargo test --doc --all -j6
+      - name: Test examples
+        run: cargo test -p examples --examples -j6
+      - name: Compile check remaining examples
+        # No OS specific code in examples, run this on fastest executor
+        if: matrix.os == 'ubuntu-latest'
+        run: cargo build -j4 -p examples
+
+      # Feature tests
+      - name: Test kube with features native-tls,ws,oauth
+        run: cargo test -p kube --lib --no-default-features --features=native-tls,ws,oauth
+        if: matrix.os == 'ubuntu-latest'
+      - name: Test kube with features rustls-tls,ws,oauth
+        run: cargo test -p kube --lib --no-default-features --features=rustls-tls,ws,oauth
+        if: matrix.os == 'ubuntu-latest'
+      # Feature tests in examples
+      - name: Test crd_derive_no_schema example
+        run: cargo test -p examples --example crd_derive_no_schema --no-default-features --features=native-tls
+        if: matrix.os == 'ubuntu-latest'
+
+  integration:
+    # Integration tests are linux only
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry/index
+            ~/.cargo/registry/cache
+            ~/.cargo/git
+            target
+          key: musl-cargo-${{ hashFiles('**/Cargo.toml') }}
+
+      - uses: nolar/setup-k3d-k3s@v1
+        with:
+          version: v1.20
+          # k3d-kube
+          k3d-name: kube
+          # Used to avoid rate limits when fetching the releases from k3s repo.
+          # Anonymous access is limited to 60 requests / hour / worker
+          # github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Compile dapp
+        run: |
+          mkdir -p ~/.cargo/{git,registry}
+          docker run --rm -t \
+              --mount type=bind,source=${{ github.workspace }},target=/volume \
+              --mount type=bind,source=$HOME/.cargo/registry,target=/root/.cargo/registry \
+              --mount type=bind,source=$HOME/.cargo/git,target=/root/.cargo/git \
+              clux/muslrust:stable \
+              cargo build -p tests --release -v
```

Comment:
i am considering just renaming this `tests` folder to `integration` as well, so that it won't be confused with all our other tests and the upcoming `kube-test`